File: Sources/LocalWhisperFlow/Services/SoundService.swift
import AppKit
import Foundation

@MainActor
final class SoundService {
    static let shared = SoundService()

    var isEnabled: Bool = true

    func playStartListening() {
        guard isEnabled else { return }
        play(named: "Tink")
    }

    func playStopListening() {
        guard isEnabled else { return }
        play(named: "Pop")
    }

    private func play(named name: String) {
        guard let sound = NSSound(named: NSSound.Name(name)) else { return }
        sound.volume = 0.4
        sound.play()
    }
}

File: Windows/LocalWhisper.Windows/Services/SoundService.cs
using System.Media;

namespace LocalWhisper.Windows.Services;

public sealed class SoundService
{
    public static SoundService Shared { get; } = new();

    private SoundService()
    {
    }

    public bool IsEnabled { get; set; } = true;

    public void PlayStartListening()
    {
        if (IsEnabled)
        {
            SystemSounds.Asterisk.Play();
        }
    }

    public void PlayStopListening()
    {
        if (IsEnabled)
        {
            SystemSounds.Beep.Play();
        }
    }
}
